src/main/java/org/shanerx/tradeshop/commands/CommandCaller.java

```diff
@@ -168,7 +189,7 @@ public boolean checkPerm() {
      * @return true if permission is NONE or sender has permission
      */
     public boolean checkPerm(TradeCommand exec) {
-        if (!cmdPass.getSender().hasPermission(exec.getPermission()) && !exec.getPermission().equals(Permissions.NONE)) {
+		if (!cmdPass.getSender().hasPermission(exec.getPermission()) && !exec.getPermission().equals(Permissions.NONE.getValue())) {
```

Comment:
A method for checking if the player has a perm using a more DRY friendly way, seeing as this same code was being run 5-10 times through the file, and none has to be checked for separately

Reply:
I am aware of that, but the .getValue
 part confuses ms

Reply:
because that is just checking if the permission enum that is being checked is NONE, because checking for the perm "" never returns true unless they literally have a blank perm

Reply:
But why is a permission check for NONE required? It literally means that no permission is required to perform that action

Reply:
Because the command enum lists Permissions.NONE when it doesnt need one otherwise that field would return null and i would still have to check for null, Permissions.NONE is just a more readable way to do it

Reply:
This was messed up a bit, the internal command check should use the enum, however the external check is just a string so thats been changed to check for "none"